diff --git a/postgres_db/src/schema.rs b/postgres_db/src/schema.rs
index b716c14..374b2c6 100644
--- a/postgres_db/src/schema.rs
+++ b/postgres_db/src/schema.rs
@@ -22,13 +22,13 @@ pub mod sql_types {
     pub struct ParsedSpecStruct;
 
     #[derive(diesel::sql_types::SqlType)]
     #[diesel(postgres_type(name = "repo_info_struct"))]
     pub struct RepoInfoStruct;
 
-    #[derive(diesel::sql_types::SqlType)]
+    #[derive(diesel::sql_types::SqlType, diesel::query_builder::QueryId)]
     #[diesel(postgres_type(name = "semver_struct"))]
     pub struct SemverStruct;
 
     #[derive(diesel::sql_types::SqlType)]
     #[diesel(postgres_type(name = "version_state"))]
     pub struct VersionState;
@@ -117,13 +117,13 @@ diesel::table! {
     use diesel::sql_types::*;
     use super::sql_types::InternalDiffLogVersionState;
 
     internal_diff_log_state (package_name) {
         package_name -> Text,
         package_only_packument_hash -> Text,
-        versions -> Array<Nullable<InternalDiffLogVersionState>>,
+        versions -> Array<InternalDiffLogVersionState>,
     }
 }
 
 diesel::table! {
     internal_state (key) {
         key -> Varchar,
@@ -138,13 +138,13 @@ diesel::table! {
     use super::sql_types::PackageState;
 
     packages (id) {
         id -> Int8,
         name -> Text,
         current_package_state_type -> PackageStateEnum,
-        package_state_history -> Array<Nullable<PackageState>>,
+        package_state_history -> Array<PackageState>,
         dist_tag_latest_version -> Nullable<Int8>,
         created -> Nullable<Timestamptz>,
         modified -> Nullable<Timestamptz>,
         other_dist_tags -> Nullable<Jsonb>,
         other_time_data -> Nullable<Jsonb>,
         unpublished_data -> Nullable<Jsonb>,
@@ -160,25 +160,26 @@ diesel::table! {
 
     versions (id) {
         id -> Int8,
         package_id -> Int8,
         semver -> SemverStruct,
         current_version_state_type -> VersionStateEnum,
-        version_state_history -> Array<Nullable<VersionState>>,
+        version_state_history -> Array<VersionState>,
         tarball_url -> Text,
         repository_raw -> Nullable<Jsonb>,
         repository_parsed -> Nullable<RepoInfoStruct>,
         created -> Timestamptz,
         extra_metadata -> Jsonb,
-        prod_dependencies -> Array<Nullable<Int8>>,
-        dev_dependencies -> Array<Nullable<Int8>>,
-        peer_dependencies -> Array<Nullable<Int8>>,
-        optional_dependencies -> Array<Nullable<Int8>>,
+        prod_dependencies -> Array<Int8>,
+        dev_dependencies -> Array<Int8>,
+        peer_dependencies -> Array<Int8>,
+        optional_dependencies -> Array<Int8>,
     }
 }
 
+diesel::joinable!(dependencies -> packages (dst_package_id_if_exists));
 diesel::joinable!(diff_log -> change_log (seq));
 
 diesel::allow_tables_to_appear_in_same_query!(
     change_log,
     dependencies,
     diff_log,
