diff --git a/postgres_db/src/schema.rs b/postgres_db/src/schema.rs
index e3e9a87..725b501 100644
--- a/postgres_db/src/schema.rs
+++ b/postgres_db/src/schema.rs
@@ -18,13 +18,13 @@ pub mod sql_types {
     pub struct ParsedSpecStruct;
 
     #[derive(diesel::sql_types::SqlType)]
     #[diesel(postgres_type(name = "repo_info_struct"))]
     pub struct RepoInfoStruct;
 
-    #[derive(diesel::sql_types::SqlType)]
+    #[derive(diesel::sql_types::SqlType, diesel::query_builder::QueryId)]
     #[diesel(postgres_type(name = "semver_struct"))]
     pub struct SemverStruct;
 }
 
 diesel::table! {
     change_log (seq) {
@@ -103,13 +103,13 @@ diesel::table! {
     use super::sql_types::InternalDiffLogVersionState;
 
     internal_diff_log_state (package_name) {
         package_name -> Text,
         package_only_packument_hash -> Text,
         deleted -> Bool,
-        versions -> Array<Nullable<InternalDiffLogVersionState>>,
+        versions -> Array<InternalDiffLogVersionState>,
     }
 }
 
 diesel::table! {
     internal_state (key) {
         key -> Varchar,
@@ -141,16 +141,16 @@ diesel::table! {
         tarball_url -> Text,
         repository_raw -> Nullable<Jsonb>,
         repository_parsed -> Nullable<RepoInfoStruct>,
         created -> Timestamptz,
         deleted -> Bool,
         extra_metadata -> Jsonb,
-        prod_dependencies -> Array<Nullable<Int8>>,
-        dev_dependencies -> Array<Nullable<Int8>>,
-        peer_dependencies -> Array<Nullable<Int8>>,
-        optional_dependencies -> Array<Nullable<Int8>>,
+        prod_dependencies -> Array<Int8>,
+        dev_dependencies -> Array<Int8>,
+        peer_dependencies -> Array<Int8>,
+        optional_dependencies -> Array<Int8>,
         secret -> Bool,
     }
 }
 
 diesel::joinable!(dependencies -> packages (dst_package_id_if_exists));
 diesel::joinable!(diff_log -> change_log (seq));
